### PR TITLE
Increase ensureGreenOnSubscriber wait timeout, fix flaky LR tests 

### PR DESCRIPTION
**Pull request by seut**
Beside of increasing the timeout, this PR includes 2 additional changes.
The first one should fix the thread leak issue of LR integration tests in case of test failures. The second is a more minor refactoring. See commits for details.

If we still see LR related test failures, we may need to investigate our CI environment as this indicates that the initial file based restore takes very long to complete.
I could reproduce the scenario by adding some thread sleeps at the beginning of the `MultiChunkTransfer.handleItems` method to simulate slow requests/IO.

### CODE REVIEW

**Comment by seut** [comment]
> I'm a bit confused about the second commit. The `join` was only called after the future already completed. Was there a problem with that?
> With the changes it's no longer waiting for the subscription state update to be through before triggering `afterReplicationStarted`. Is that intended?

Hm right, no this wasn't directly intended, thanks for catching it! My main motivation was to move future chaining to one place for better readability. But I also wonder if we really want to abort the restore if the subscription state update failed. Maybe we should add a try/catch while still waiting for the response?

**Comment by seut** [comment]
I've pushed a fixup to partly revert the 2nd commit (keep waiting for the subscription state update), I will change the commit message once approved. Please check.

**Review by mfussenegger — APPROVED** [review]
> But I also wonder if we really want to abort the restore if the subscription state update failed. Maybe we should add a try/catch while still waiting for the response?

I'm not sure but I generally wonder if we could avoid some of the explicit update if instead we inferred the state based on other information that's available.

E.g. That a subscription is restoring should be visible due to the snapshot/restore activity.

That's probably a bigger change, but would likely also help reduce load on a cluster.

**Pull request merged**
